web/src/presence.ts

```diff
@@ -177,7 +184,8 @@ export function set_info(
         }
     */
 
-    clear_internal_data();
+    presence_last_update_id = last_update_id;
```

Comment:
Flagging this change, since it's small but important. `clean_internal_data()` purges all the presence data the webapp has, so that doesn't work with getting only recent presences from the server.

Seems to me that it's fine to just delete this call, so the webapp will keep the data it already has and patch the entries in it for the users that  got updated.

Reply:
But I don't understand why this call had to be here to begin with, because if we were fetching all data then the loop below would have just overwritten everything anyway. So maybe I'm missing something.

Reply:
Digging with `git log -S'user_info = {}'  --follow -- static/js/activity.js`, I'm pretty sure the answer is there's no real reason; 5d4b2305fe0aee979fb241cd3b53e022ab107ff3 is what introduced this, and it looks like basically just an artifact of past refactoring.